tasks/pkg-macos.yml

```diff
@@ -17,70 +17,76 @@
     msg: "The Datadog ansible role wasn't able to find the user : {{ datadog_macos_user }}"
   when: mac_user_check.rc != 0
 
-- include_tasks: pkg-macos/macos_agent_latest.yml
+- name: Include macos agent latest version install task
+  include_tasks: pkg-macos/macos_agent_latest.yml
   when: (not datadog_skip_install) and (datadog_agent_macos_version is not defined)
 
-- include_tasks: pkg-macos/macos_agent_version.yml
+- name: Include macos agent pinned version install task
+  include_tasks: pkg-macos/macos_agent_version.yml
   when: (not datadog_skip_install) and (datadog_agent_macos_version is defined)
 
 - name: Display macOS download URL
   debug:
     var: dd_download_url
   when: not datadog_skip_install
 
-- name: pre-Delete temporary dmg
+- name: Pre-Delete temporary dmg
   file:
-    path: '/tmp/datadog-agent.dmg'
+    path: /tmp/datadog-agent.dmg
     state: absent
-  become: yes
+  become: true
   when: not datadog_skip_install
 
 - name: Create temporary datadog install user file
   copy:
-    dest: "/tmp/datadog-install-user"
+    dest: /tmp/datadog-install-user
     content: "{{ datadog_macos_user }}"
-    mode: 0554
+    mode: "0554"
   when: (not datadog_skip_install) and (not ansible_check_mode)
 
 - name: Download macOS datadog agent
   get_url:
     url: "{{ dd_download_url }}"
-    dest: '/tmp/datadog-agent.dmg'
-    mode: 0750
+    dest: /tmp/datadog-agent.dmg
+    mode: "0750"
   register: download_dmg_result
   when: (not datadog_skip_install) and (not ansible_check_mode)
 
 - name: Detach agent dmg if already mounted
-  shell: 'hdiutil detach "/Volumes/datadog_agent" >/dev/null 2>&1 || true'
+  shell: hdiutil detach "/Volumes/datadog_agent" >/dev/null 2>&1 || true
   when: (not datadog_skip_install) and (not ansible_check_mode)
+  changed_when: true
 
 - name: Attach agent dmg
-  command: 'hdiutil attach /tmp/datadog-agent.dmg -mountpoint "/Volumes/datadog_agent"'
+  command: hdiutil attach /tmp/datadog-agent.dmg -mountpoint "/Volumes/datadog_agent"
   when: (not datadog_skip_install) and (not ansible_check_mode) and (download_dmg_result.status_code == 200)
+  changed_when: true
 
 - name: Unpack and copy Datadog Agent files
   shell:
-    cmd: '/usr/sbin/installer -pkg "`find "/Volumes/datadog_agent" -name \*.pkg 2>/dev/null`" -target /'
-    chdir: '/'
-  become: yes
+    cmd: /usr/sbin/installer -pkg "`find "/Volumes/datadog_agent" -name \*.pkg 2>/dev/null`" -target /
+    chdir: /
+  become: true
   register: datadog_agent_install
   when: (not datadog_skip_install) and (not ansible_check_mode) and (download_dmg_result.status_code == 200)
+  changed_when: true
```

Comment:
Why does ansible-lint want us to use `changed_when: true` on this task and the above ones?

Reply:
Ok, so I reached the conclusion that the `changed_when: true` in this file are actually ok. These tasks will be skipped if `datadog_skip_install` is set to `true`, but otherwise they will actually do things that (possibly) change things on the host (and then they'll trigger the macos handlers to restart the agent, which again will actually execute commands which restart the agent).